src/gc/gc.cpp

```diff
@@ -2963,8 +2966,12 @@ gc_heap::dt_low_ephemeral_space_p (gc_tuning_point tp)
             
             dprintf (GTC_LOG, ("h%d: plan eph size is %Id, new gen0 is %Id", 
                 heap_number, plan_ephemeral_size, new_gen0size));
-            ret = ((size_t)(heap_segment_reserved (ephemeral_heap_segment) - (heap_segment_mem (ephemeral_heap_segment))) <
-                    (plan_ephemeral_size + new_gen0size));
+
+            // While we were in the no gc region we could have allocated a larger than normal segment,
```

Comment:
> While we were in the no gc region [...]

This sounds like we are assuming that we know we got here after being in a no-GC region. Is that always the case?

I just did a quick search and it looks like we only call `dt_low_ephemeral_space_p` with `tuning_deciding_promote_ephemeral` if the pause mode is _not_ `pause_no_gc` (in `gc_heap::expand_heap`). I think I'm missing something here.